caikit/runtime/http_server/pydantic_wrapper.py

```diff
@@ -170,3 +181,182 @@ def _from_base64(data: Union[bytes, str]) -> bytes:
     if isinstance(data, str):
         return base64.b64decode(data.encode("utf-8"))
     return data
+
+
+async def pydantic_from_request(
+    pydantic_model: Type[pydantic.BaseModel], request: Request
+):
+    """Function to convert a fastapi request into a given pydantic model. This
+    function parses the requests Content-Type and then correctly decodes the data.
+    The currently supported Content-Types are `application/json`
+    and `multipart/form-data`"""
+    content_type = request.headers.get("Content-Type")
+    log.debug4("Detected request using %s type", content_type)
```

Comment:
NIT: We probably want this on a more visible level since I suspect this could be very useful and will only log once per request.

```suggestion
    log.debug("Detected request using %s type", content_type)
```

Reply:
Good call! I wasn't sure what the level scheme was for caikit so I just did the lowest I found.